### ISSUE TITLE
No attached drives

### ISSUE THREAD

**Issue by ajonesist** [thomas-krenn/check_lsi_raid#2]
Hello, 
This is a very useful plugin, thanks!
Just one issue...
I've been testing this with our systems, and have found that in a scenario where the storage is disconnected from the controller, the plugin reports 'Invalid StorCLI command!' (UNKNOWN status).

Details below from test system - 'MegaRAID SAS 2108' with no shelf attached:

`./check_lsi_raid -C 0`
_Invalid StorCLI command! (/usr/bin/storcli64 /c0/eall/sall show all)_

`echo $?`
_3_

`/usr/bin/storcli64 /c0/eall/sall show all`
_Controller = 0
Status = Failure
Description = No drive _found!_

It would be nice to have this default to a warning status, or even better to have it a configurable option so that if there are no disk drives present, the plugin could be configured to handle this as OK / WARN or CRIT.

Thanks again
Aled

**Comment by gschoenberger** [comment]
Thanks for the opened issue!
Would it be sufficient to just exit with "WARNING" if there are no attached drives?

I think it would be easier to just exit as the status checks and performance data formatting rely on present physical disks.

**Comment by ajonesist** [comment]
Yes, exit with WARNING status would be perfectly sufficient.
For us, the controller should always have disks attached except during maintenance, in which case the host would be in downtime anyway. So a warning instead of unknown would be great.

Thanks 
Aled

**Comment by gschoenberger** [comment]
Could you try latest Master with commit 877027a ?
It should exit with warning now if no drives are found.

THX, Georg

**Comment by ajonesist** [comment]
Tested on host with drives attached and host with no drives attached.
With no drives attached, I get the 'no storage attached' message and a status 1 Warning.
With drives attached I get the correct status, same as before.

I did have to change the string that it grep's for though, 
` if(grep { /No drive _found/i } @output){`
to
` if(grep { /No drive found/i } @output){`

(removing the underscore), must have been a typo from my side when I pasted the details here.

So with that underscore removed, its perfect.

Thanks,
Aled
